docs/app/pages/components/components-sections/date-time-picker/date-range-picker/date-range-picker.component.ts

```diff
@@ -78,7 +78,7 @@ export class ComponentsDateRangePickerComponent extends BaseDocumentationSection
         // check if the date contains a hyphen
         const parts = (date.indexOf('—') ? date.split('—') : date.split('-'));
         const startDate = Date.parse(parts[0].trim());
-        const endDate = Date.parse(parts[0].trim());
+        const endDate = Date.parse(parts[1].trim());
 
         if (parts.length >= 1 && !isNaN(startDate)) {
```

Comment:
Checking parts.length here and below is redundant after already doing `parts[i].trim()` above. Array bounds ought to be checked before accessing the items.